lte/gateway/python/integ_tests/s1aptests/test_no_attach_complete_with_mme_restart.py

```diff
@@ -117,17 +117,7 @@ def test_no_attach_complete_with_mme_restart(self):
             time.sleep(1)
 
         # Receive NW initiated detach request
-        response = self._s1ap_wrapper.s1_util.get_response()
-
-        while response.msg_type == s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND:
-            print(
-                "Received Attach Accept retransmission from before restart",
-                "Ignoring...",
-            )
-            response = self._s1ap_wrapper.s1_util.get_response()
-
-        self.assertEqual(
-            response.msg_type,
+        response = self._s1ap_wrapper.s1_util.get_response(
```

Comment:
This fix for this test case has some technical issues like get_response() does not expect the message type as argument and the function assertResponseIgnoringAttachAccept is not utilized. Moreover, I tested this fix and this test case still seems to be flaky. 

The current issue with the test case is that it handles the retransmitted NW initiated detach only once whereas MME is able to send it multiple times because it comes up early after restart. Please have a look on the fixed test case I have attached here:
[test_no_attach_complete_with_mme_restart.py.txt](https://github.com/magma/magma/files/9299655/test_no_attach_complete_with_mme_restart.py.txt)

Reply:
Thank you, wow, I must have mixed something up in my commit ... `assertResponseIgnoringAttachAccept` should have been used in all calls after the mme restart to filter out possible `UE_ATTACH_ACCEPT_IND` responses. Sorry for that.

But your solution seems more sophisticated so I applied it (with minor comment changes). Is this OK for you?